indexer/indexer.go

```diff
@@ -233,14 +236,23 @@ func ProcessDataFeed(csConsumer transports.Consumer, txFeed *txfeed.TxFeed, db *
 				megaStatement := []string{}
 				megaParameters := []interface{}{}
 				for _, pb := range chainUpdate.Added() {
-					for _, indexer := range indexers {
-						s, err := indexer.Index(pb)
-						log.Debug("inside indexer loop", "idx", indexer, "len", len(s))
-						if err != nil {
-							log.Error("Error computing updates", "err", err.Error())
-							continue UPDATELOOP
+					for k, v := range pb.Values {
+						switch k {
+						case safeNumKey:
+							safeNum = new(big.Int).SetBytes(v)
+						case finalizedNumKey:
+							finalizedNum = new(big.Int).SetBytes(v)
+						default:
+							for _, indexer := range indexers {
```

Comment:
This is going to result in iterating over each indexer for each key in pb.Values, which is going to make the megastatement huge and also wrong.

We don't actually need to iterate over pb.Values here. That's a common pattern when we have to see if the value matches a regular expression (and I may have done that in Cardinal EVM in a place where were already iterating over all the keys), but in this case we could just do something like:

```
if v, ok := pb.Values[safeNumKey]; ok {
  safeNum = new(big.Int).SetBytes(v)
}
if v, ok := pb.Values[finalizedNumKey]; ok {
  finalizedNum = new(big.Int).SetBytes(v)
}
```

And never have to iterate over the values.